File: src/main/java/com/flowcommerce/core/domain/member/Seller.java
package com.flowcommerce.core.domain.member;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Seller {
    private Long id;
    private Long memberId;
    private String sellerName;
    private SellerStatus status;

    private CompanyInfo companyInfo;
    private ContactInfo contactInfo;
    private PaymentInfo paymentInfo;
    private Commission commission;

    private Long approvedBy;
    private LocalDateTime approvedAt;
}

File: src/main/kotlin/com/flowcommerce/core/domain/member/Seller.kt
package com.flowcommerce.core.domain.member

import java.time.LocalDateTime

data class Seller(
    val id: Long?,
    val memberId: Long,
    val sellerName: String,
    val status: SellerStatus,
    val companyInfo: CompanyInfo,
    val contactInfo: ContactInfo,
    val paymentInfo: PaymentInfo,
    val commission: Commission,
    val approvedBy: Long?,
    val approvedAt: LocalDateTime?
)
